src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,102 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const areFieldsFilled = () => {
+    if (title !== '' && imgUrl !== '' && imdbUrl !== '' && imdbId !== '') {
+      return true;
+    }
+
+    return false;
+  };
+
+  const areEmptyFields = title !== '' && imgUrl !== '' && imdbUrl !== ''
+  && imdbId !== '';
```

Comment:
the variable name does not reproduce the logic. The variable says that it is a future value that checks if all your variables are empty, and you write the code itself if all your variables are not empty

```suggestion
  const areEmptyFields = Boolean(!title && !imgUrl && !imdbUrl && !imdbId);
```